ext/gcc/newdelete_cortex.cpp

```diff
@@ -53,15 +60,15 @@ void *
 operator new(size_t size, modm::MemoryTraits traits) noexcept
 {
 	void * ptr = malloc_tr(size, traits.value);
-	modm_assert(ptr, "core", "heap", "new", size);
+	assert_ptr(ptr, size);
 	return ptr;
 }
 
 void *
 operator new[](size_t size, modm::MemoryTraits traits) noexcept
```

Comment:
This should not be marked `noexcept`, I will add a separate version with `nothrow` overload

Reply:
I'll add it in a separate PR that deals with refactoring heap implementation.